### PR TITLE
Add GitHub Actions for CI and release build

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis (OpenFaaS Ltd) <alexellis2@gmail.com>

Adds a build for Go and a release upload as per the arkade and k3sup project.

Feel free to take this as an initial commit for a Linux binary, and then add other cross-compilation options when you have time.

Tested on my fork: 

* CI build - https://github.com/alexellis/nerdctl/actions/runs/410243071
* Release build and artifact - https://github.com/alexellis/nerdctl/actions/runs/410245763 - https://github.com/alexellis/nerdctl/releases/tag/0.0.1

Screenshot 2020-12-09 at 09 23 11

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<!--- Provide a general summary of your changes in the Title above -->
```
Comment (AkihiroSuda) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Thanks, but I'd like this template to be separate from this PR. I'm not sure I want to add checklist.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    - '*'
+  pull_request:
+    branches:
+    - '*'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        go-version: [1.13.x]
```
Comment (AkihiroSuda) [inline .github/workflows/build.yml:15]: 1.15

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@master
```
Comment (AkihiroSuda) [inline .github/workflows/build.yml:19]: v2

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  publish:
+    strategy:
+      matrix:
+        go-version: [ 1.13.x ]
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@master
```
Comment (AkihiroSuda) [inline .github/workflows/publish.yml:16]: v2

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  pull_request:
+    branches:
+    - '*'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        go-version: [1.13.x]
+        os: [ubuntu-latest]
```
Comment (AkihiroSuda) [inline .github/workflows/build.yml:16]: ubuntu-20.04

**Comment by alexellis** [comment]
Do you just want to merge this and make it look how you want? You know just how you like things, my intention is only to help you by sharing our (tested) GitHub actions recipes

**Comment by alexellis** [comment]
All changes done.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by alexellis** [comment]
No problems.

If you can cut something like a 0.1.0 release, folks can start downloading binaries etc.

**Comment by AkihiroSuda** [comment]
uploaded binaries https://github.com/AkihiroSuda/nerdctl/releases

**Pull request merged**